contracts/upgradeable_contracts/arbitrary_message/MessageDelivery.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity 0.4.24;
+
+import "../../libraries/SafeMath.sol";
+
+
+contract MessageDelivery {
+    using SafeMath for uint256;
+
+    function requireToPassMessage(address _contract, bytes _data, uint256 _gas) public {
+        require(isMessageDeliverySubsidizedMode());
+        require(_gas >= getMinimumGasUsage(_data) && _gas <= getMaxGasPerTx());
+        emitEventOnMessageRequest(abi.encodePacked(msg.sender, _contract, _gas, uint8(0x00), _data));
+    }
+
+    function requireToPassMessage(address _contract, bytes _data, uint256 _gas, uint256 _gasPrice) public {
```

Comment:
@akolotov Right now there is no check for `gasPrice` specified by the user. There is a potential issue where gas price could be very low and so when the transaction is send to the other network by the validator, it may not be processed and also will affect the next transactions sent by the validator because of the nonce.

One approach to prevent this issue, could be to introduce a new parameter `minGasPrice` which can be the same value as `gasPrice` of the other network.
This parameter could be set on `initialize` method and we can add the following check on `requireToPassMessage`:
```
require(_gasPrice >= minGasPrice());
```

Does it makes sense? Do you have any other ideas?